src/harness/testcases/WINNF_FT_S_SIQ_testcase.py

```diff
@@ -841,6 +842,237 @@ def test_WINNF_FT_S_SIQ_11(self):
+
+    # N2 spectrum inquiry requests
+    # 1. Spectrum Inquiry: 3550-3580 .
+    spectrum_inquiry_1 = json.load(
+        open(os.path.join('testcases', 'testdata', 'spectrum_inquiry_0.json')))
+    spectrum_inquiry_1['inquiredSpectrum'] = [{
+        'lowFrequency': 3550000000,
+        'highFrequency': 3580000000
+    }]
+
+    # 2. Spectrum Inquiry: Invalid Value of low and high frequency and invalid id for cbsdId.
+    spectrum_inquiry_2 = json.load(
+        open(os.path.join('testcases', 'testdata', 'spectrum_inquiry_0.json')))
+    spectrum_inquiry_2['inquiredSpectrum'] = [{
+        'lowFrequency': 3610000000,
+        'highFrequency': 3590000000
+    }]
+    spectrum_inquiry_2['cbsdId'] = "INVALID_CBSD_ID_12345"
+
+    # 3. Spectrum Inquiry: Missing parameter low frequency and missing cbsdId
+    spectrum_inquiry_3 = json.load(
+        open(os.path.join('testcases', 'testdata', 'spectrum_inquiry_0.json')))
+    spectrum_inquiry_3['inquiredSpectrum'] = [{
+        'highFrequency': 3700000000
+    }]
+    spectrum_inquiry_3['cbsdId'] = "REMOVE"
+
+    # Create the actual config.
+    devices = [device_b, device_c, device_e]
+    conditionals = {
+        'registrationData': [conditionalParameters]
+    }
+
+    config = {
+        'registrationRequestsN2': devices,
+        'conditionalRegistrationData' : conditionals,
+        'expectedResponseCodes': [(0,), (103,), (102,)],  
+        'palRecordsN3':[pal_records1,pal_records2],
+        'ppaRecordsN3':[{'ppaRecord':ppa_record1,
+                         'ppaClusterList': ppa_cluster_list_1},
+                        {'ppaRecord':ppa_record2,
+                         'ppaClusterList': ppa_cluster_list_2}],
+   
+        'spectrumInquiryRequestsN2':[spectrum_inquiry_1,spectrum_inquiry_2,spectrum_inquiry_3],
+        'gwpzRecordsN1':[gwpz_e],
+        'fr1Cbsd': [ [{'frequency' :{'lowFrequency': 3570000000,'highFrequency': 3580000000}} 
+                       ],[],[] ], 
+
+        'fr2Cbsd': [ [ {'frequency' :{'lowFrequency': 3570000000,'highFrequency': 3580000000}}],
+                       [],[] ]
+
+    }
+    writeConfig(filename, config)
+
+  @configurable_testcase(generate_SIQ_12_default_config)
+  def test_WINNF_FT_S_SIQ_12(self, config_filename):
+    """[Configurable] Spectrum inquiry in the presence of protected entities."""
+
+    config = loadConfig(config_filename)
+        
+    # Light checking of the config file
+    self.assertEqual(len(config['spectrumInquiryRequestsN2']),len(config['registrationRequestsN2']))
+    self.assertEqual(len(config['expectedResponseCodes']),len(config['spectrumInquiryRequestsN2']))
+    self.assertEqual(len(config['spectrumInquiryRequestsN2']),len(config['fr1Cbsd']))
+    self.assertEqual(len(config['spectrumInquiryRequestsN2']),len(config['fr2Cbsd']))
+    self.assertGreater(len(config['registrationRequestsN2']),0)
+
+    # Step1: Load information about N1 GWBZs
+    if ('gwpzRecordsN1' in config) and (config['gwpzRecordsN1']):
```

Comment:
You can assume that this is included in the config, although it may have zero records. Same for all of the other fields except conditionalRegistrationData.

Reply:
Removed the extraneous check and assume this field is there, even if there are zero records.